# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/webrender_bindings/RenderTextureHostSWGL.cpp
# Commit: 91d8f0f829fd
# Full Hash: 91d8f0f829fd2ae6ec234a6a3f372bf503ed2fbd
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2020-11-06 04:14:34
# Regressor Bug: 1673983
# File Overlap Count: 2
# Description:
#   Bug 1673983 - Add RenderTextureHostSWGL support to D3D11 RenderTextureHosts. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D95799
# ==============================================================================

diff -r 91cb8ee8ddd8 -r 91d8f0f829fd gfx/webrender_bindings/RenderTextureHostSWGL.cpp
--- a/gfx/webrender_bindings/RenderTextureHostSWGL.cpp	Thu Nov 05 19:26:21 2020 +0000
+++ b/gfx/webrender_bindings/RenderTextureHostSWGL.cpp	Thu Nov 05 20:18:13 2020 +0000
@@ -10,7 +10,8 @@
 namespace mozilla {
 namespace wr {
 
-bool RenderTextureHostSWGL::UpdatePlanes(wr::ImageRendering aRendering) {
+bool RenderTextureHostSWGL::UpdatePlanes(RenderCompositor* aCompositor,
+                                         wr::ImageRendering aRendering) {
   wr_swgl_make_current(mContext);
   size_t planeCount = GetPlaneCount();
   bool filterUpdate = IsFilterUpdateNecessary(aRendering);
@@ -25,7 +26,7 @@
   gfx::ColorDepth colorDepth = GetColorDepth();
   for (size_t i = 0; i < planeCount; i++) {
     PlaneInfo& plane = mPlanes[i];
-    if (!MapPlane(i, plane)) {
+    if (!MapPlane(aCompositor, i, plane)) {
       if (i > 0) {
         UnmapPlanes();
       }
@@ -94,12 +95,13 @@
 }
 
 wr::WrExternalImage RenderTextureHostSWGL::LockSWGL(
-    uint8_t aChannelIndex, void* aContext, wr::ImageRendering aRendering) {
+    uint8_t aChannelIndex, void* aContext, RenderCompositor* aCompositor,
+    wr::ImageRendering aRendering) {
   if (!SetContext(aContext)) {
     return InvalidToWrExternalImage();
   }
   if (!mLocked) {
-    if (!UpdatePlanes(aRendering)) {
+    if (!UpdatePlanes(aCompositor, aRendering)) {
       return InvalidToWrExternalImage();
     }
     mLocked = true;
@@ -142,7 +144,7 @@
     return false;
   }
   if (!mLocked) {
-    if (!UpdatePlanes(mCachedRendering)) {
+    if (!UpdatePlanes(nullptr, mCachedRendering)) {
       return false;
     }
     mLocked = true;